### PR TITLE
Add fire()/kick() to event manager

### PR DESCRIPTION
**Pull request by jiangliu**
When integrating with the event-manager, we found several enhancements.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -137,6 +137,46 @@ impl<T: EventSubscriber + ?Sized> MutEventSubscriber for T {
...
+    fn process(&mut self, events: Events, ops: &mut EventOps) {
+        self.deref_mut().process(events, ops);
+    }
+
+    fn init(&mut self, ops: &mut EventOps) {
+        self.deref_mut().init(ops);
+    }
+}
+
+impl MutEventSubscriber for Box<dyn MutEventSubscriber + Send> {
```
Comment (andreeaflorescu) [inline src/lib.rs:153]: Does this need to be explicitly implemented?

**Comment by andreeaflorescu** [comment]
@jiangliu can you please open the PR from a fork branch? Buildkite seems to have problems when opening PRs from a branch that is in the same repository. The status of the PR cannot be shown anymore. I would also like us to not pollute the repository with PR branches.

File: src/lib.rs [hunk 2]
```diff
@@ -50,9 +50,18 @@ pub enum Error {
 pub type Result<T> = result::Result<T, Error>;
 
 /// Opaque object that uniquely represents a subscriber registered with an `EventManager`.
-#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Hash)]
```
[the remark this answers is not in the document]
Comment (jiangliu) [reply src/lib.rs:56]: The default invalid design helps to avoid using Option<SubscriberId>:)

File: src/lib.rs (hunk 1 above)
Comment (jiangliu) [reply src/lib.rs:153]: Yes, the boxed series of implementations are added on demand.

File: src/lib.rs [hunk 3]
```diff
@@ -12,6 +12,9 @@ use std::sync::{Arc, Mutex};
 
 use vmm_sys_util::errno::Error as Errno;
 
+/// Reexport EventSet from vmm_sys_util::epoll.
```
[the remark this answers is not in the document]
Comment (jiangliu) [reply src/lib.rs:15]: Fixed

**Comment by jiangliu** [comment]
> @jiangliu can you please open the PR from a fork branch? Buildkite seems to have problems when opening PRs from a branch that is in the same repository. The status of the PR cannot be shown anymore. I would also like us to not pollute the repository with PR branches.

Sorry, I made a mistake when dealing with vmm-epoll/event-manager history:(
Will close this PR and open another one from forked repo.

**Pull request closed without merging**